bundles/org.openhab.binding.airq/src/main/java/org/openhab/binding/airq/internal/AirqHandler.java

```diff
@@ -0,0 +1,797 @@
+                                dec.get(airqName).toString());
+                    }
+                    break;
+                case "nightmode":
+                    JsonElement daynightdata = new Gson().fromJson(dec.get(airqName).toString(), JsonElement.class);
+                    if (daynightdata != null) {
+                        JsonObject json_daynightdata = daynightdata.getAsJsonObject();
+                        processType(json_daynightdata, "StartDay", "nightMode_startDay", "string");
+                        processType(json_daynightdata, "StartNight", "nightMode_startNight", "string");
+                        processType(json_daynightdata, "BrightnessDay", "nightMode_brightnessDay", "number");
+                        processType(json_daynightdata, "BrightnessNight", "nightMode_brightnessNight", "number");
+                        processType(json_daynightdata, "FanNightOff", "nightMode_fanNightOff", "boolean");
+                        processType(json_daynightdata, "WifiNightOff", "nightMode_wifiNightOff", "boolean");
+                    } else {
+                        logger.warn("air-Q - airqHandler - processType(): Cannot extract day/night data: {}",
+                                dec.get(airqName).toString());
+                    }
+                    break;
+                case "wlan":
+                    JsonElement wlandata = new Gson().fromJson(dec.get(airqName).toString(), JsonElement.class);
+                    if (wlandata != null) {
+                        JsonObject json_wlandata = wlandata.getAsJsonObject();
+                        processType(json_wlandata, "Gateway", "wlanConfigGateway", "string");
+                        processType(json_wlandata, "MAC", "wlanConfigMac", "string");
+                        processType(json_wlandata, "SSID", "wlanConfigSsid", "string");
+                        processType(json_wlandata, "IP address", "wlanConfigIPAddress", "string");
+                        processType(json_wlandata, "Net Mask", "wlanConfigNetMask", "string");
+                        processType(json_wlandata, "BSSID", "wlanConfigBssid", "string");
+                    } else {
+                        logger.warn(
+                                "air-Q - airqHandler - processType(): Cannot extract WLAN data from this string: {}",
+                                dec.get(airqName).toString());
+                    }
+                    break;
+                case "arr":
+                    JsonElement jsonarr = new Gson().fromJson(dec.get(airqName).toString(), JsonElement.class);
+                    if ((jsonarr != null) && (jsonarr.isJsonArray())) {
+                        JsonArray arr = jsonarr.getAsJsonArray();
+                        String str = new String();
+                        for (JsonElement el : arr) {
+                            str = str.concat(el.getAsString()).concat(", ");
+                        }
```

Comment:
When building Strings in loops, you could use `StringBuilder` to save resources.